### DISCUSSION TITLE
Enable "terminal monitor"

### DISCUSSION THREAD

**Discussion by amitsova** [ktbyers/netmiko#3304] [Q&A] [answered]
Hello everyone,

Does anyone have an experience enabling terminal monitor? My colleagues want it enabled in my scripts, but after doing some tests, I noted that when it is enabled, very often the scripts fail with ReadTimeout exception (unable to find different patterns - from config terminal to show outputs. I increased the read_timeout value to 40 seconds (so far the default is 10, I think), but this did not help much.
I would be really thankful if someone can give me some hints.

Thank you in advance!

**Comment by ktbyers** [comment, marked as answer]
Yes, `terminal monitor` would generally be a problem (as it will emit things to output that will mess up the script).

Why do your colleagues want it enabled?

It would probably be better to capture log messages in a different way and not have it cause issues with the scripting.

**Reply by amitsova** [reply]
Hi Kirk,

Thanks for your reply! Well, in the beginning it was the "% Access VLAN does not exist. Creating vlan 18" message (when you add an access port to a non-existing vlan). However, I realized that for this you do not need "terminal monitor", it just pops-up on the screen. Then we started thinking to catch messages when an ether-channel member is kicked out due to some config, or hsrp changes, for example.
But anyway, I will just try to catch it with regex in the syslog ("show log") output,  I think this should work fine.

Thanks a lot again!

Kind regards,
Adriana
